main/integration-tests/__test__/api-tests/workspace-types/get-workspace-types.test.js

```diff
@@ -0,0 +1,93 @@
+ *  A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+const { runSetup } = require('../../../support/setup');
+const errorCode = require('../../../support/utils/error-code');
+
+describe('Get workspace-types scenarios', () => {
+  let setup;
+  let adminSession;
+
+  beforeAll(async () => {
+    setup = await runSetup();
+    adminSession = await setup.defaultAdminSession();
+  });
+
+  afterAll(async () => {
+    await setup.cleanup();
+  });
+
+  describe('Get workspace-types', () => {
+    it('should fail to get approved if user is inactive', async () => {
+      const adminSession2 = await setup.createAdminSession();
+      await adminSession2.resources.users.deactivateUser(adminSession2.user);
+
+      await expect(adminSession2.resources.workspaceTypes.getApproved()).rejects.toMatchObject({
+        code: errorCode.http.code.unauthorized,
+      });
+    });
+
+    it('should fail to get not-approved if user is inactive', async () => {
+      const adminSession2 = await setup.createAdminSession();
+      await adminSession2.resources.users.deactivateUser(adminSession2.user);
+
+      await expect(adminSession2.resources.workspaceTypes.getNotApproved()).rejects.toMatchObject({
+        code: errorCode.http.code.unauthorized,
+      });
+    });
+
+    it('should fail to get approved if user is anonymous', async () => {
+      const anonymousSession = await setup.createAnonymousSession();
+
+      await expect(anonymousSession.resources.workspaceTypes.getApproved()).rejects.toMatchObject({
+        code: errorCode.http.code.badImplementation,
+      });
+    });
+
+    it('should fail to get not-approved if user is anonymous', async () => {
+      const anonymousSession = await setup.createAnonymousSession();
+
+      await expect(anonymousSession.resources.workspaceTypes.getNotApproved()).rejects.toMatchObject({
+        code: errorCode.http.code.badImplementation,
+      });
+    });
+
+    it('should return approved if user is researcher', async () => {
+      const researcherSession = await setup.createResearcherSession();
+
+      await expect(researcherSession.resources.workspaceTypes.getApproved()).resolves.toEqual(expect.any(Array));
+    });
+
+    it('should return no not-approved if user is researcher', async () => {
+      const researcherSession = await setup.createResearcherSession();
+      const workspaceTypeId = setup.gen.string({ prefix: 'workspace-test' });
+
+      await adminSession.resources.workspaceTypes.create({
+        id: workspaceTypeId,
+        status: 'not-approved',
+      });
+
+      await expect(researcherSession.resources.workspaceTypes.getNotApproved()).resolves.toEqual([]);
```

Comment:
Can we add not-approved workspace type so the return is not empty?

Reply:
Thanks, I have updated this test case as well as other positive test cases to create the appropriate workspace type in case one does not already exist in the database. Note, I have also updated the logic to check the array.  If I expect the array to be empty, I call "toHaveLength(0)". If I expect the array to not be empty, I call "not.toHaveLength(0)"